source/camera/CullingManager.cpp

```diff
@@ -0,0 +1,289 @@
+            LogManager::getSingleton().logMessage( ss.str(),LogMessageLevel::NORMAL);
+            ss.str("");
+        }
+    
+        int64_t mm = ((std::min(xxLeft, xxLeftOld) >> mPrecisionDigits) << mPrecisionDigits) ;
+        if(std::min(xxLeft, xxLeftOld) < max(xxRight,xxRightOld) )
+            for (int64_t xx = mm ; xx <= max(xxRight,xxRightOld) ; xx+= Unit)
+            {
+                bool bash = (xx >= xxLeftOld && xx <= xxRightOld && (yy >= oldWalk.getBottomLeftVertex().y) && yy <= oldWalk.getTopLeftVertex().y);
+                bool splash = (xx >= xxLeft && xx <= xxRight && (yy >= mWalk.getBottomLeftVertex().y) && yy <= mWalk.getTopLeftVertex().y);
+
+                if (mDebug)
+                {
+                    ss<< " x " <<  (xx >> mPrecisionDigits )<< " y " << (yy >> mPrecisionDigits ) << " " << (splash && (mode & SHOW)) << (bash && (mode & HIDE)) << endl;
+            
+                }
+                xxp = ((xx >>mPrecisionDigits) - 128.0);
+                yyp = ((yy >>mPrecisionDigits) - 128.0);
+                GameMap* gm = mCm->mGameMap;
+                if(bash && splash && (mode & HIDE) && (mode & SHOW))
+                {
+                    // Nothing
+                }
+
+                else if (gm && bash && (mode & HIDE) && xxp>=0 && yyp>= 0 && xxp<mCm->mGameMap->getMapSizeX() && yyp < mCm->mGameMap->getMapSizeY())
+                    gm->getTile(xxp, yyp)->hide();
+                else if (gm && splash && (mode & SHOW) && xxp>=0 && yyp>= 0 && xxp<mCm->mGameMap->getMapSizeX() && yyp < mCm->mGameMap->getMapSizeY())
+                    gm->getTile(xxp, yyp)->show();
+            }
+
+    }
+}
+
+bool CullingManager::getIntersectionPoints()
+{
+    const Ogre::Vector3* myvector = mCm->getActiveCamera()->getWorldSpaceCorners();
+
+    mMyRay[0]= Ogre::Ray (myvector[0], myvector[4] - myvector[0]);
+    mMyRay[1]= Ogre::Ray (myvector[1], myvector[5] - myvector[1]);
+    mMyRay[2]= Ogre::Ray (myvector[2], myvector[6] - myvector[2]);
+    mMyRay[3]= Ogre::Ray (myvector[3], myvector[7] - myvector[3]);
+
+    std::pair<bool, Ogre::Real> intersectionResult;
+
+    for(int ii = 0; ii < 4; ++ii)
+    {
+        intersectionResult =  mMyRay[ii].intersects(mMyplanes);
+        if(intersectionResult.first)
+            mOgreVectorsArray[ii]= (mMyRay[ii].getPoint(intersectionResult.second));
+        if(intersectionResult.first)
+        {
+        }
+        else
+        {
+            cerr<< "I didn't find the intersection point for " << ii <<"th ray"<<endl;
+            exit(1);
+        }
+    }
+    return true;
+}
+
+bool CullingManager::onFrameStarted()
+{
+    if(mCullTilesFlag)
+        getIntersectionPoints();
+    if(mCullTilesFlag)
+        cullTiles();
+
+    return true;
+}
+
+bool CullingManager::onFrameEnded()
+{
+    return true;
+}
+
+/*! \brief Sort two Vector3i p1 and p2  to satisfy p1 <= p2 according to
```

Comment:
Eh eh, of course not. But this is not a comment, this is a function description.